legate/core/runtime.py

```diff
@@ -880,7 +1286,273 @@ def dispatch(self, op, redop=None):
+                offsets.append(i+1)
+                if input.kind is Future:
+                    future_start+=1
+            for o,output in enumerate(op._outputs):
+                offsets.append(-(o+1)) 
+            for r,reduction in enumerate(op._reductions):
+                offsets.append(-(r+1)) 
+ 
+            op_ids.append(numpy_context.get_task_id(op._task_id._value_))
+            offset_start+=(len(op._inputs)+len(op._outputs))
+            input_start+=len(op._inputs)
+            output_start+=len(op._outputs)
+            reduction_start+=len(op._reductions)
+            scalar_start+=len(op._scalar_args)
+
+        #terminators
+        input_starts.append(input_start)
+        output_starts.append(output_start)
+        offset_starts.append(offset_start)
+        reduction_starts.append(reduction_start)
+        scalar_starts.append(scalar_start)
+        future_starts.append(future_start)
+
+        meta_arrs =  (input_starts, output_starts, offset_starts, offsets, reduction_starts,  scalar_starts, 
+                      future_starts, op_ids)
+        fusion_metadata = FusionMetadata(*meta_arrs)
+        meta_maps=None
+        return meta_maps, fusion_metadata
+   
+
+    def build_fused_op(self,ops):
+
+        for i in range(len(ops)):
+            self.propogateFuture(ops[i])
+        fusion_checker = FusionChecker(ops, self._contexts, self)
+        fusion_checker.register_constraint(AllValidOps())
+        fusion_checker.register_constraint(IdenticalLaunchShapes())
+        fusion_checker.register_constraint(IdenticalProjection())
+        fusion_checker.register_constraint(ValidProducerConsumer())
+        can_fuse,fusable_sets, partitions = fusion_checker.can_fuse()
+
+        #short circuit         
+        #if not can_fuse:
+        ##    drint("CANNOT FUSE!")
+        #    return False, partitions
+                
+        super_strats = []
+        super_fspaces = []
+        super_strategies = []
+        super_keystores = [] 
+        z=0
+        for fusable_set in fusable_sets:   
+            #create super strategy for this fusable set
+            super_strat = {}
+            super_fspace = {}
+            super_keystore = set()
+            start,end = fusable_set
+            drint("creating fusable set for", start, end)
+            for j in range(start,end):
+                super_strat = {**(super_strat.copy()), **partitions[j]._strategy}
+                super_fspace = {**(super_fspace.copy()), **partitions[j]._fspaces}
+                super_keystore = super_keystore.union(partitions[j]._key_parts)
+            super_strats.append(super_strat)
+            super_fspaces.append(super_fspace)
+            super_keystores.append(super_keystore)
+            super_strategies.append(Strategy(partitions[start]._launch_shape, super_strat, super_fspace, super_keystore))
+        #drint("lens", len(super_strats), len(super_fspaces), len(super_strategies), len(super_keystore))
+       
```

Comment:
creates the "super-partition" by aggregating every unfused operations partitions/stratgies